### PR TITLE
Feature: namespace_selector support for having multiple values for the same label

### PR DESCRIPTION
**Pull request by vortegatorres**
### Type of change:

Adding `namespace_selector` support for having multiple values for the same label. Context https://github.com/apache/apisix-ingress-controller/issues/1987

- [ ] Bugfix
- [X] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [ ] Refactor
- [ ] Chore
- [ ] CI/CD or Tests

### What this PR does / why we need it:

In the `namespace_selector` configuration, when there are multiple values for the same label, only the last one is taken into account. This happens because the label key serves as the unique identifier [in this map](https://github.com/apache/apisix-ingress-controller/blob/bca0d1da5e0d721ff7c94555821db55dd2659504/pkg/providers/k8s/namespace/namespace_provider.go#L70), and the most recent value will replace any previous ones.

For instance, if you have the following values, only "app3" would be considered in the list of watched namespaces:
```
namespace_selector:
        - "kubernetes.io/metadata.name=app1"
        - "kubernetes.io/metadata.name=app2"
        - "kubernetes.io/metadata.name=app3" 
```

### Pre-submission checklist:

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: pkg/providers/k8s/namespace/namespace_provider.go [hunk 1]
```diff
@@ -103,7 +103,21 @@ func (c *watchingProvider) Init(ctx context.Context) error {
...
+	for key, values := range valuesByKey {
+		requirement := metav1.LabelSelectorRequirement{
+			Key:      key,
+			Operator: metav1.LabelSelectorOpIn,
+			Values:   values,
+		}
+		matchExpressions = append(matchExpressions, requirement)
+	}
+
+	labelSelector := metav1.LabelSelector{MatchExpressions: matchExpressions}
```
Comment (vortegatorres) [inline pkg/providers/k8s/namespace/namespace_provider.go:120]: I opted for `MatchLabels` over `MatchExpressions` in the LabelSelector criteria. This choice enables us to select multiple label values and facilitates support for a `namespace_selector` with multiple values for the same label: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements

**Review by tao12345666333 — COMMENTED** [review]
Thanks 

Please add an e2e test case to cover this feature.

**Comment by vortegatorres** [comment]
> Please add an e2e test case to cover this feature.

Where is the best place to add the test?

**Comment by vortegatorres** [comment]
> Thanks
> 
> Please add an e2e test case to cover this feature.

@tao12345666333 I added the e2e test case in `test/e2e/suite-ingress/suite-ingress-features/namespace.go`. Please let me know if it looks good to you.

**Comment by vortegatorres** [comment]
@tao12345666333 What should we do next to get this pull request merged?

**Comment by vortegatorres** [comment]
>  @vortegatorres Please merge the master branch . Let's make the CI happy first

Done!

**Comment by tao12345666333** [comment]
please check failed jobs

**Comment by vortegatorres** [comment]
> please check failed jobs

Apart from the e2e tests, which, looking at other PRs, seems generally failing, the only failing job is the `unit-test-ci`. However, I am unable to replicate any of the `unit-test-ci` issues running the unit tests. Is there a way to rerun it? I don't believe there is any problem with it.

**Comment by tao12345666333** [comment]
let me re-run it

**Comment by tao12345666333** [comment]
It looks like this changes some behavior of the APISIX Ingress. I don't have time to deal with this recently, maybe @AlinsRan  can take a look

**Comment by lingsamuel** [comment]
I ran some failed tests (`suite-chore/endpoints.go`) locally and they passed normally. That's weird. I wll rerun this file to see if anything different.

**Comment by AlinsRan** [comment]
Let me check it.

**Comment by Revolyssup** [comment]
Closing this PR. I will create another one

**Comment by vortegatorres** [comment]
> Closing this PR. I will create another one

@Revolyssup is it not possible to keep the same changes in this PR?

**Pull request closed without merging**